manim/utils/tex.py

```diff
@@ -1,298 +1,179 @@
-        self.rebuild_cache()
-        pass
+            self.preamble += "\n" + txt
+        self.rebuild()
 
-    def clear_preamble(self):
-        """Removes custom definitions from the LaTeX preamble.
-        This does not affect the imported packages or documentclass."""
-        self.common_preamble_text = ""
-        self.ctex_preamble_text = ""
-        self.tex_preamble_text = ""
-        self.rebuild_cache()
-        pass
-
-    def generate_tex_command(self, command, *, required_params, optional_params=[]):
-        """
-        Function for creating LaTeX command strings with or without options.
-        Internally used to generate ``\\usepackage`` commands.
+    def add_to_document(self, txt):
+        """Adds txt to the TeX template just after \\begin{document}, e.g. ``\\boldmath``
 
         Parameters
         ----------
-        command : :class:`str`
-            The command, e.g. ``"usepackage"``.
-        required_params : Iterable[:class:`str`]
-            The required parameters of this command, each wrapped in ``{}``.
-        optional_params : Iterable[:class:`str`]
-             The optional parameters of this command, each separated by a comma inside one ``[]``.
-
-        Examples
-        --------
-        ::
-
-            generate_tex_command("usepackage", required_params=["packagename"], optional_params=["option1", "option2"])
-
-        Returns
-        -------
-        :class:`str`
-            The generated command.
+        txt : :class:`str`
+            String containing the text to be added.
         """
-        optional_params = list(optional_params)  # so we can measure its length
-        return r"\{0}{1}{2}".format(
-            command,
-            f"[{','.join(optional_params)}]" if optional_params else "",
-            "".join("{" + param + "}" for param in required_params),
-        )
+        self.post_doc_commands += "\n" + txt + "\n"
+        self.rebuild()
 
-    def generate_usepackage(self, pkg):
-        if isinstance(pkg, list):
-            return self.generate_tex_command(
-                "usepackage", required_params=[pkg[0]], optional_params=pkg[1]
-            )
-        else:
-            return self.generate_tex_command("usepackage", required_params=[pkg])
-
-    def get_text_for_text_mode(self, expression):
-        """Inserts an expression verbatim into the TeX template.
+    def get_texcode_for_expression(self, expression):
+        """Inserts expression verbatim into TeX template.
 
         Parameters
         ----------
         expression : :class:`str`
-            The expression to be typeset, e.g. ``"foo"``.
+            The string containing the expression to be typeset, e.g. ``$\\sqrt{2}$``
 
         Returns
         -------
         :class:`str`
-            LaTeX code based on the template, containing the given expression and ready for typesetting
+            LaTeX code based on template, containing the given expression and ready for typesetting
```

Comment:
why was "the" removed here? Lol
Either "the" or "a", pick one

Reply:
"template" here is the argument to the function.

LaTeX code based on **template**, containing the given **expression** and ready for typesetting

Reply:
and it wasn't removed per se. The whole file was written from scratch.. it's just github that insists on showing this as a series of diffs.

Reply:
Is there some qualifier similar to `:attr:` so we could write "based on `:param:template` or something like that?

Reply:
Ahhh I see, then just surround "template" by double-backticks (\`\`template\`\`) so that users become aware that you refer to the parameter.

Reply:
actually, I think I was mistaken here.The template in question is `self`. I have reworded it.